src/content/blog/astro-570.mdx

```diff
@@ -0,0 +1,235 @@
+```
+
+They can be used in Astro components, pages, API endpoints, actions, and middleware. The API is the same for all of them:
+
+```ts title="src/actions/addToCart.ts" {9,10}
+import { defineAction } from "astro:actions";
+import { z } from "astro:schema";
+export const server = {
+  addToCart: defineAction({
+    input: z.object({
+      product: z.string(),
+    }),
+    async handler({ product }, { session }) {
+      const cart = await session.get('cart');
+      session.set('cart', [...cart, product]);
+    },
+  }),
+};
+```
+
+### Type-safety
+
+Of course, as this is Astro, your sessions can be type-safe too. Define an interface and you will get type-checking and auto-completion for all of your session data:
+
+
+```ts title="src/env.d.ts"
+declare namespace App {
+  interface SessionData {
+    user: {
+      id: string;
+      name: string;
+    };
+    cart: string[];
+  }
+}
+```
+
+### Cookie-less sessions
+
+The session ID is automatically tracked using a cookie, but you can also use them without cookies by using the `session.load()` method. This is useful for things like APIs, where you might want to use a custom header or params to track the session ID:
+
+```ts title="src/pages/api/cart.ts" "session.load(sessionId)"
+export async function POST({ session, request }: APIContext) {
+  const sessionId = request.headers.get('x-session-id');
+
+  // If there is an existing session ID, load it. Otherwise one will be created
+  if (sessionId) {
+    await session.load(sessionId);
+  } 
+
+  const cart = await session.get('cart');
+  const data = await request.json<{ item: string }>();
+
+  if (!data?.item) {
+    return new Response('Item is required', { status: 400 });
+  }
+  session.set('cart', [...cart, data.item]);
+  // Return the cart and session ID
+  return Response.json({ cart, sessionId: session.sessionId });
+}
+```
+
+
+### Storage drivers
+
+Astro sessions use storage drivers to persist the data, and these are either zero-config or require minimal setup for all official Astro adapters. Alternatively you can use any [unstorage](https://unstorage.unjs.io/) driver, which supports dozens of storage backends such as Redis and MongoDB.
+
+- **[Node](https://docs.astro.build/en/guides/integrations-guide/node/#sessions)**: zero-config filesystem driver
+- **[Netlify](https://docs.astro.build/en/guides/integrations-guide/netlify/#sessions)**: zero-config Netlify Blobs driver
+- **[Cloudflare](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#sessions)**: minimal-config Workers KV driver
+- **[Vercel](https://docs.astro.build/en/guides/integrations-guide/vercel/#sessions)**: minimal-config Redis/Upstash driver
+
+On any of these platforms you can easily switch out the driver for a different one if you don't want to use the default.
+
+```js title="astro.config.mjs" {5-10}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  // ...
+  session: {
```

Comment:
```suggestion
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
  // ...
  adapter: netlify(),
  session: {
```
I think this does a few things:
- reinforces that an adapter is needed (and that `server` output is not!)
- shows an official adapter configured with a non-default storage system
- is a nice nod to Netlify 😄